ra2ce/analyses/indirect/losses.py

```diff
@@ -125,91 +161,145 @@ def calc_vlh(
+                            * values_of_time["business"]["value_of_time"]
+                    )
+                    + (
+                            traffic_data["evening_other"]
+                            / traffic_data["evening_total"]
+                            * values_of_time["other"]["value_of_time"]
+                    )
             )
             # to calculate costs per unit traffi per hour. This is weighted based on the traffic mix and value of each traffic type
         vlh["euro_vlh"] = vlh["euro_per_hour"] * vlh["vlh_total"]
         return vlh
 
-    def calculate_losses_from_table(self):
-        """This function opens an existing table with traffic data and value of time to calculate losses based on detouring values. It also includes
-        a traffic jam estimation.
-        #TODO: check if gdf already exists from effectiveness measures.
-        #TODO: If not: read feather file.
-        #TODO: if yes: read gdf
-        #TODO: koppelen van VVU aan de directe schade berekeningen
-        """
-        traffic_data = self.load_df(self.losses_input_path, "traffic_intensities.csv")
-        dict1 = {
-            "AS_VTG": "evening_total",
-            "AS_FRGT": "evening_freight",
-            "AS_COMM": "evening_commute",
-            "AS_BUSS": "evening_business",
-            "AS_OTHR": "evening_other",
-            "ET_FRGT": "day_freight",
-            "ET_COMM": "day_commute",
-            "ET_BUSS": "day_business",
-            "ET_OTHR": "day_other",
-            "ET_VTG": "day_total",
-            "afstand": "distance",
-            "H_Cap": "capacity",
-            "H_Stroken": "lanes",
-        }
-        traffic_data.rename(columns=dict1, inplace=True)
-
-        detour_data = self.load_df(_losses_input_path, "detour_data.csv")
-        dict2 = {
-            "VA_AV_HWN": "detour_time_evening",
-            "VA_RD_HWN": "detour_time_remaining",
-            "VA_OS_HWN": "detour_time_morning",
-            "VA_Etm_HWN": "detour_time_day",
-        }
-        detour_data.rename(columns=dict2, inplace=True)
-
-        vehicle_loss_hours = self.vehicle_loss_hours(_losses_input_path)
-        vlh = self.calc_vlh(traffic_data, vehicle_loss_hours, detour_data)
+    def calc_vlh(self) -> pd.DataFrame:
+        def _get_range(height: float) -> str:
+            for range_tuple in self.inundation_height_ranges:
+                x, y = range_tuple
+                if x <= height <= y:
+                    return f"{x}_{y}"
+            raise ValueError(f"No matching range found for height {height}")
+
+        def _set_vot_intensity_per_trip_purpose(purposes: list):
+            for purpose in purposes:
+                vot_var_name = f"vot_{purpose}"
+                partofday_trip_purpose_name = self.partofday + f"_{purpose}"
+                partofday_trip_purpose_intensity_name = "intensity_" + partofday_trip_purpose_name
+                # read and set the vot's
+                globals()[vot_var_name] = \
```

Comment:
when and why are globals introduced? We want to get rid of these!

Reply:
I wanted to set and update variables with varying variable name.